linkage-checker-exclusion-rules.xml

```diff
@@ -0,0 +1,28 @@
+<LinkageCheckerFilter>
+  <LinkageError>
+    <Source><Package name="com.nimbusds.jose"/></Source>
+    <Reason>Optional</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Target><Package name="org.conscrypt"/></Target>
+    <Reason>Optional</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Target><Package name="com.sun.jna"/></Target>
+    <Reason>Optional</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Target><Package name="org.tukaani.xz"/></Target>
+    <Reason>Optional</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Source><Package name="org.apache.hadoop.mapred"/></Source>
```

Comment:
@sfc-gh-pbennes Do you think if we ok to ignore the following 2 linkage issues:
conflict of hadoop-common
snowflake-jdbc shade

Reply:
snowflake-jdbc issues need to be also fixed but as an intermediate step they can be ignored for now

Reply:
Yes, problems with shading and relocation @sfc-gh-lsembera AFAICT. I think someone's going to work on addressing these in follow-up PRs.